app/controllers/agent.js

```diff
@@ -28,4 +28,13 @@ export default Ember.Controller.extend(DetailMixin, {
     outgoingAffiliations: Ember.computed('model.outgoingAgentRelations', function() {
         return this.get('model.outgoingAgentRelations').map(relation => relation.related);
     }),
+
+    actions: {
+        goBack() {
+            if (document.referrer.includes('\/share')) {
+                history.back();
+            }
+            this.transitionToRoute('discover');
```

Comment:
I don't think they can both be hit because if `history.back()` is called, the page transitions. Is that what you're saying? That both `history.back()` and `this.transitionToRoute()` could be called?